packages/core/src/libraries/verification.ts

```diff
@@ -27,17 +43,86 @@ export const buildUserVerificationRecordById = async (
   return buildVerificationRecord(libraries, queries, result.data);
 };
 
-export const saveVerificationRecord = async (
-  userId: string,
+/**
+ * Verifies the user sensitive permission by checking if the verification record is valid
+ * and associated with the user.
+ */
+export const verifyUserSensitivePermission = async ({
+  userId,
+  id,
+  queries,
+  libraries,
+}: {
+  userId: string;
+  id: string;
+  queries: Queries;
+  libraries: Libraries;
+}): Promise<void> => {
+  try {
+    const record = await buildVerificationRecordById({ id, queries, libraries, userId });
+
+    assertThat(record.isVerified, 'verification_record.not_found');
+  } catch (error) {
+    if (error instanceof RequestError) {
+      throw new RequestError({ code: 'verification_record.permission_denied', status: 403 });
```

Comment:
Should be 401? need to re-verify the user's identity.

Reply:
updated